### PR TITLE
Fix "namespace \"karmada-cluster"\ not found" when cluster is being registered with "pull" mode

### PR DESCRIPTION
**Pull request by nuclearwu**
Fix "namespace \"karmada-cluster"\ not found" when cluster is being registered with "pull" mode

Signed-off-by: wuzhongjian <wuzhongjian_yewu@cmss.chinamobile.com>

**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:
After installed karmada through "kubectl karmada init", and then registering the member cluster with ‘pull’ mode, the agent log reports an error ”namespace \"karmada-cluster"\ not found“，so the ”karmada-cluster“ namespace needs to be created by default when the karmada control plane is initialized.
45dac71f0e9384104fc36a03aa080b8
b7e01e82d3b5a93c2c62b932c9b6eaa

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

**Comment by nuclearwu** [comment]
Most people use karmada to add clusters through push mode, and then consider adding clusters through pull mode, but the karmada-cluster namespace is created by default only when clusters are added through push mode, so we have not found this problem. Including the official documents of karmada, it is also said that two clusters are added through the push mode first, and then a cluster is added through the pull mode. So when you add a cluster directly through pull mode, you will find that the karmada-cluster namespace is not created by default, which will cause this error.

File: pkg/karmadactl/cmdinit/karmada/deploy.go [hunk 1]
```diff
@@ -4,6 +4,7 @@ import (
 	"context"
 	"encoding/json"
 	"fmt"
+	options2 "github.com/karmada-io/karmada/pkg/karmadactl/options"
```
Comment (wuyingjun-lucky) [inline pkg/karmadactl/cmdinit/karmada/deploy.go:7]: We organize the import package in three parts:
`go package

third-party package

current project package`

**Comment by wuyingjun-lucky** [comment]
@RainbowMango  help to active the pipeline

**Comment by wuyingjun-lucky** [comment]
@nuclearwu 
You can use 
[squash](https://github.com/karmada-io/karmada/blob/master/docs/contributors/guide/github-workflow.md#squash-commits) to squash commits

**Comment by nuclearwu** [comment]
https://github.com/karmada-io/karmada/pull/2328 @wuyingjun-lucky

**Comment by RainbowMango** [comment]
@nuclearwu You don't need to close and reopen a new one. Just following the GitHub workflow would be fine.
https://karmada.io/docs/next/contributor/github-workflow

**Pull request closed without merging**